bin/prover-client/src/proving_ops/btc_ops.rs

```diff
@@ -25,19 +18,23 @@ use crate::{
 /// Operations required for BTC block proving tasks.
 #[derive(Debug, Clone)]
 pub struct BtcOperations {
+    rollup_params: Arc<RollupParams>,
     btc_client: Arc<BitcoinClient>,
 }
 
 impl BtcOperations {
     /// Creates a new BTC operations instance.
-    pub fn new(btc_client: Arc<BitcoinClient>) -> Self {
-        Self { btc_client }
+    pub fn new(btc_client: Arc<BitcoinClient>, rollup_params: Arc<RollupParams>) -> Self {
+        Self {
+            btc_client,
+            rollup_params,
+        }
     }
 }
 
 #[async_trait]
 impl ProvingOperations for BtcOperations {
-    type Input = (Block, RollupParams);
+    type Input = (Block, Vec<TxFilterRule>, CredRule);
```

Comment:
Should probably wrap this in a struct at this point.

Reply:
Will work on a better interface for handling this. Planning to move this into `proof-impl` crate for better consistency.